_posts/intjnotinfp/2020-09-03-first-reflection-intjnotinfp.md

```diff
@@ -0,0 +1,27 @@
+---
+layout: post
+author: intjnotinfp
+title: "intjnotinfp's first reflection"
+---
+
+This module was confidence building but I’m hoping it’s not a false sense of confidence because I can tell how easy the assignments were. I liked doing the assigned readings and working through things alone and not under the pressure to “get” it quickly in class like the circles and turtles exercises last week. It's useful to note that I like learning this stuff individually then bringing questions or concerns to class afterward. Hoonestly, I can't keep up in class when we fly through examples. The readings were interesting, and I like that Dr. Hauser identified why we’re starting with Python (i.e. the easier syntax, usability, and fact that it can be used in many different contexts. Also, in regard to the comments sections in the Chapter 2 Variables Notes, someone who works in tech told me that no one really uses comments with the pound sign – is that because that’s the “advanced” way to do things? I’m finding “commenting things out” to be pretty useful, so I’m going to keep using that feature regardless. I’m feeling optimistic that I can learn this stuff if it’s broken down in small chunks like this. I was really surprised at how good it feels to finally figure some piece of code out, especially after you’ve been struggling with it for a little while. That feeling when you finally “get” something is pretty great. My programmer friends have told me that it never goes away. It’s kind of like a runner’s high.. but better.
```

Comment:
Your friend in tech was, I hope, referring to a more advanced way of leaving comments in Python, the docstring. We'll cover that later in class, but it's basically an easy way to leave multi-line comments, and there are tools to auotmatically generate documentation from docstrings, as opposed to single-line comments with `#`.

Regardless, comments are critical in any language. How , when, and in what format to leave them can vary by project type. That's one of the many things you'll have to adapt to when joining a new project or job. But, regardless, in this class you'll learn how to leave good comments for yourself and others.